core/domain/exp_services_test.py

```diff
@@ -1305,6 +1305,49 @@ def test_unpublished_explorations_can_accept_commit_message(self):
             ), None)
 
 
+class UpdateExplorationContributionDatetimeTests(ExplorationServicesUnitTests):
+    """Test whether contribution date changes with publication of exploration
+    and update of already published exploration
+    """
+
+    def setUp(self):
```

Comment:
Addressed the first case.

I believe the second case might be too complex to address. If A makes a commit to Exploration 1, their first commit, that is later unpublished, but has since made commits to Exploration 2 and Exploration 3, how would we change the exploration date? I believe our philosophy in the past has been that any sort of publication or commit, no matter if it is later deleted, counts as a first contribution.